gematria/datasets/process_and_filter_bbs_lib_test.cc

```diff
@@ -0,0 +1,162 @@
+// Copyright 2024 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gematria/datasets/process_and_filter_bbs_lib.h"
+
+#include "X86InstrInfo.h"
+#include "absl/log/check.h"
+#include "gematria/llvm/asm_parser.h"
+#include "gematria/llvm/disassembler.h"
+#include "gematria/llvm/llvm_architecture_support.h"
+#include "gematria/utils/string.h"
+#include "gtest/gtest.h"
+#include "llvm/MC/MCCodeEmitter.h"
+
+using namespace llvm;
+
+namespace gematria {
+namespace {
+
+class ProcessFilterBBsTest : public testing::Test {
+ private:
+  std::unique_ptr<LlvmArchitectureSupport> LLVMArchSupport;
+  std::unique_ptr<MCInstPrinter> LLVMInstPrinter;
+  BBProcessorFilter BBProcessor;
+
+ protected:
+  ProcessFilterBBsTest() : BBProcessor(){};
+
+  void SetUp() override {
+    LLVMArchSupport = LlvmArchitectureSupport::X86_64();
+    LLVMInstPrinter =
+        LLVMArchSupport->CreateMCInstPrinter(InlineAsm::AsmDialect::AD_ATT);
```

Comment:
Optional: These could be initialized inline (GoogleTest creates a new instance for each test fixture anyway).